sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/telemetry/Telemetry.scala

```diff
@@ -36,6 +37,14 @@ object Telemetry extends ExtensionId[Telemetry] {
   val TRACE_ID: String = "trace_id"
   override def createExtension(system: ExtendedActorSystem): Telemetry =
     new Telemetry(system)
+
+  private val traceParentPattern: Regex = """^[0-9a-f]{2}-([0-9a-f]{32})-[0-9a-f]{16}-[0-9a-f]{2}$""".r
+
+  def extractTraceId(traceParent: String): String =
+    traceParent match {
+      case traceParentPattern(traceId) => traceId
+      case _                           => ""
```

Comment:
What is the scenario where this would not be passed a string matches? Isn't that an error rather than something that should return an empty trace id?

Reply:
Good point. Are you suggesting to throw an exception here? If we throw an exception and at any point we add the wrong data to the metadata, as `traceparent` we could break the application. 
I would rather log no trace found in this `traceparent` and return the blank. WDYT?

Reply:
Or return option if you _think_ it could be present but not contain trace id, but then would that ever be valid? Also, is there no built in opentelemetry thing for extracting it rather than doing our own string manipulation?

Reply:
it seems they do this when invalid.  returning `00000000000000000000000000000000`
https://github.com/open-telemetry/opentelemetry-java/blob/dcdc6727f78638557ccc65742cb7898e306ea227/api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java#L78

That makes more sense

Reply:
so how about returning "00000000000000000000000000000000"?

Reply:
From TraceId.java
> A class that represents a trace identifier. A valid trace identifier is a 16-byte array with at least one non-zero byte.

Isn't it possible to reach it via `Metadata#traceContext#asOpenTelemetryContext#something`?

Reply:
you're right. 
```
Span.fromContext(context).getSpanContext.getTraceId
```

Reply:
As suggested, we can do the same with the following:
```
  def extractTraceId(metadata: Metadata): String = {
    val context = W3CTraceContextPropagator.getInstance().extract(Context.root(), metadata, otelGetter)
    Span.fromContext(context).getSpanContext.getTraceId
  }
``` 

When no traceparent return is `00000000000000000000000000000000`

Reply:
I have mixed feelings. I think this is more obscure (the first line of `extractTraceId` specifically)  but I like the signature more. `Metadata` as input instead of a `String`. That said I could keep this signature with the other implementation. 
In any case I see some benefits in this implementation in that there is less risk I got something wrong. Nor that there was much to get wrong in the RegEx but anyways. How you find it?

Reply:
The SpanContext is already available from MetadataImpl, so no need to parse the w3c format again just for logging?

`def traceId(): String = Span.fromContext(asOpenTelemetryContext()).getSpanContext.getTraceId` in the SDK `MetadataImpl#traceContext` seems reasonable to me.

Reply:
You mean, use it in the `Telemetry` method? or to add this method to our `TraceContext` 
I like the latter.